test/es6/dynamic-module-functionality.js

```diff
@@ -0,0 +1,359 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+// ES6 Module functionality tests -- verifies functionality of import and export statements
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+function testScript(source, message, shouldFail = false, explicitAsync = false) {
+    message += " (script)";
+    let testfunc = () => testRunner.LoadScript(source, undefined, shouldFail, explicitAsync);
+
+    if (shouldFail) {
+        let caught = false;
+
+        assert.throws(testfunc, SyntaxErrr, message);
+        assert.isTrue(caught, `Expected error not thrown: ${message}`);
+    } else {
+        assert.doesNotThrow(testfunc, message);
+    }
+}
+
+function testModuleScript(source, message, shouldFail = false, explicitAsync = false) {
+    message += " (module)";
+    let testfunc = () => testRunner.LoadModule(source, 'samethread', shouldFail, explicitAsync);
+
+    if (shouldFail) {
+        let caught = false;
+
+        // We can't use assert.throws here because the SyntaxError used to construct the thrown error
+        // is from a different context so it won't be strictly equal to our SyntaxError.
+        try {
+            testfunc();
+        } catch(e) {
+            caught = true;
+
+            // Compare toString output of SyntaxError and other context SyntaxError constructor.
+            assert.areEqual(e.constructor.toString(), SyntaxError.toString(), message);
+        }
+
+        assert.isTrue(caught, `Expected error not thrown: ${message}`);
+    } else {
+        assert.doesNotThrow(testfunc, message);
+    }
+}
+
+function testDynamicImport(importFunc, thenFunc, catchFunc, _asyncEnter, _asyncExit) {
+    var promise = importFunc();
+    assert.isTrue(promise instanceof Promise);
+    promise.then((v)=>{
+        _asyncEnter();
+        thenFunc(v);
+        _asyncExit();
+    }).catch((err)=>{
+        _asyncEnter();
+        catchFunc(err);
+        _asyncExit();
+    });
+}
+
+var tests = [
+    {
+        name: "Runtime evaluation of module specifier",
+        body: function () {
+            let functionBody =
+                `testDynamicImport(
+                    ()=>{
+                        var getName = ()=>{ return 'ModuleSimpleExport'; };
+                        return import( getName() + '.js');
```

Comment:
>import [](start = 31, length = 6)

What if a non-string is passed to import? like import() or import({}) etc. Can we add test cases covering error scenarios?